src/editors/containers/ProblemEditor/components/SelectTypeModal/hooks.test.js

```diff
@@ -0,0 +1,152 @@
+import React from 'react';
+import { MockUseState } from '../../../../../testUtils';
+import * as module from './hooks';
+import { ProblemTypeKeys } from '../../../../data/constants/problem';
+
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useState: (val) => ({ useState: val }),
+  useEffect: jest.fn(),
+}));
+
+const state = new MockUseState(module);
+const mockSetProblemType = jest.fn().mockName('setProblemType');
+const mockSelected = 'vAl';
+const mockSetSelected = jest.fn().mockName('setSelected');
+
+let hook;
+
+describe('SelectTypeModal hooks', () => {
+  beforeEach(() => {
+    state.mock();
+  });
+  afterEach(() => {
+    state.restore();
+    jest.clearAllMocks();
+  });
+
+  describe('selectHooks', () => {
+    beforeEach(() => {
+      hook = module.selectHooks();
+    })
+    test('selected defaults to SINGLESELECT', () => {
+      expect(hook.selected).toEqual(ProblemTypeKeys.SINGLESELECT);
+    });
+    test('setSelected sets state as expected', () => {
+      const expectedArg = 'neWvAl'
+      state.mockVal(state.keys.selected, 'mOcKvAl');
+      hook.setSelected(expectedArg);
+      expect(state.setState.selected).toHaveBeenCalledWith(expectedArg);
+    });
+  });
+
+  describe('onSelect', () => {
+    test('setProblemType is called with selected', () => {
+      module.onSelect(mockSetProblemType, mockSelected)();
+      expect(mockSetProblemType).toHaveBeenCalledWith({ selected: mockSelected });
+    });
+  });
+
+  describe('useArrowNav', () => {
```

Comment:
@connorhaugh I managed to write these tests which work. However it's definitely not cleaning up the useEffects, which could cause issues with other tests in the future.